packages/php/blueprint/src/Importers/ImportInstallPlugin.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+namespace Automattic\WooCommerce\Blueprint\Importers;
+
+use Automattic\WooCommerce\Blueprint\ResourceStorages;
+use Automattic\WooCommerce\Blueprint\StepProcessor;
+use Automattic\WooCommerce\Blueprint\StepProcessorResult;
+use Automattic\WooCommerce\Blueprint\Steps\InstallPlugin;
+use Plugin_Upgrader;
+
+class ImportInstallPlugin implements StepProcessor {
+	private ResourceStorages $storage;
+	private array $installed_plugin_paths = array();
+
+	public function __construct( ResourceStorages $storage ) {
+		$this->storage = $storage;
+	}
+	public function process( $schema ): StepProcessorResult {
+		$result = StepProcessorResult::success( 'InstallPlugins' );
+
+		$installed_plugins = $this->get_installed_plugins_paths();
+		$plugin            = $schema->pluginZipFile;
+
+		if ( isset( $installed_plugins[ $plugin->slug ] ) ) {
+			$result->add_info( "Skipped installing {$plugin->slug}. It is already installed." );
+			return $result;
+		}
+		if ( $this->storage->is_supported_resource( $plugin->resource ) === false ) {
+			$result->add_error( "Invalid resource type for {$plugin->slug}." );
+			return $result;
+		}
+
+		$downloaded_path = $this->storage->download( $plugin->slug, $plugin->resource );
+		if ( ! $downloaded_path ) {
+			$result->add_error( "Unable to download {$plugin->slug} with {$plugin->resource} resource type." );
+			return $result;
+		}
+
+		$install = $this->install( $downloaded_path );
+		$install && $result->add_info( "Installed {$plugin->slug}." );
+
+		if ( isset( $plugin->options, $plugin->options->activate ) && $plugin->options->activate === true ) {
+			$activate = $this->activate( $plugin->slug );
+			if ( $activate ) {
```

Comment:
The `activate_plugin` returns null on success and a WP_Error object on failure. 

https://developer.wordpress.org/reference/functions/activate_plugin/

Reply:
Thank you!

Fixed in [Fix plugin activation condition](https://github.com/woocommerce/woocommerce/pull/49763/commits/eed86c0fb854734dc15d2bc80d79637e6b57e389)